### PR TITLE
entgql: multi order support using annotation

### PR DESCRIPTION
**Pull request by a8m**

### CODE REVIEW

File: entgql/pagination.go [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+	}
+	return OrderDirectionDesc
+}
+
+// PageInfo of a connection type.
+type PageInfo[T any] struct {
+	HasNextPage     bool       `json:"hasNextPage"`
+	HasPreviousPage bool       `json:"hasPreviousPage"`
+	StartCursor     *Cursor[T] `json:"startCursor"`
+	EndCursor       *Cursor[T] `json:"endCursor"`
```
Comment (giautm) [inline entgql/pagination.go:84]:
```suggestion
	StartCursor     T `json:"startCursor"`
	EndCursor       T `json:"endCursor"`
```

File: entgql/template/pagination.tmpl [hunk 2]
```diff
@@ -32,139 +32,25 @@ import (
...
-
-const (
-	// OrderDirectionAsc specifies an ascending order.
-	OrderDirectionAsc OrderDirection  = "ASC"
-	// OrderDirectionDesc specifies a descending order.
-	OrderDirectionDesc OrderDirection = "DESC"
+// Common entgql types.
+type (
+	Cursor = entgql.Cursor[{{ $idType }}]
+	PageInfo = entgql.PageInfo[{{ $idType }}]
```
Comment (giautm) [inline entgql/template/pagination.tmpl:45]:
```suggestion
	PageInfo = entgql.PageInfo[*Cursor]
```

**Review by giautm — APPROVED** [review]
💯

File: entgql/pagination.go (hunk 1 above)
Comment (a8m) [reply entgql/pagination.go:84]: I wanted to emphasis that this is limited to `Cursor`s. Don't you think so?

File: entgql/template/pagination.tmpl (hunk 2 above)
Comment (a8m) [reply entgql/template/pagination.tmpl:45]: ditto

**Pull request merged**